pandas/tests/indexes/test_threadsafety.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+"""
+Testing that indexes (and its engine) work correclty with multiple threads
+"""
+
+import pytest
+import pandas as pd
+
+from concurrent.futures import ThreadPoolExecutor
+
+
+class TestThreadSafety(object):
+
+    @pytest.mark.slow
+    @pytest.mark.parametrize('execution_number', range(7))
+    def test_isunique(self, execution_number):
+        """This test is executed seven times, each time it uses a pool of
+        two threads to run a test that is very likely to fail without the
+        fix for gh-21150. It is not a deterministic test, as there is
+        still a chance it will pass even though the bug exists. But
+        with the fix, it must always work with not issues."""
+        x = pd.date_range('2001', '2020')
+        with ThreadPoolExecutor(2) as p:
+            assert all(p.map(lambda x: x.is_unique, [x] * 2))
```

Comment:
Hmm...I think we can incorporate this test into one of the existing files in this directory.

Reply:
Which one you think would accommodate this one? Happy to update the pull request with this change :-)

Reply:
`test_base.py` should suffice.

Reply:
Done!